datafusion/proto/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -756,6 +769,82 @@ impl LogicalExtensionCodec for TopKExtensionCodec {
     }
 }
 
+#[derive(Debug)]
+pub struct ScalarUDFExtensionCodec {}
+
+impl LogicalExtensionCodec for ScalarUDFExtensionCodec {
+    fn try_decode(
+        &self,
+        _buf: &[u8],
+        _inputs: &[LogicalPlan],
+        _ctx: &SessionContext,
+    ) -> Result<Extension> {
+        not_impl_err!("No extension codec provided")
+    }
+
+    fn try_encode(&self, _node: &Extension, _buf: &mut Vec<u8>) -> Result<()> {
+        not_impl_err!("No extension codec provided")
+    }
+
+    fn try_decode_table_provider(
+        &self,
+        _buf: &[u8],
+        _schema: SchemaRef,
+        _ctx: &SessionContext,
+    ) -> Result<Arc<dyn TableProvider>> {
+        internal_err!("unsupported plan type")
+    }
+
+    fn try_encode_table_provider(
+        &self,
+        _node: Arc<dyn TableProvider>,
+        _buf: &mut Vec<u8>,
+    ) -> Result<()> {
+        internal_err!("unsupported plan type")
+    }
+
+    fn try_decode_udf(
+        &self,
+        _name: &str,
+        buf: &[u8],
+        ctx: &dyn FunctionRegistry,
+    ) -> Result<Arc<ScalarUDF>> {
+        let msg = ScalarUDFProto::decode(buf).map_err(|_| {
+            DataFusionError::Internal("Error decoding test table".to_string())
+        })?;
+        if let Some(expr) = msg.expr.as_ref() {
+            let node = from_proto::parse_expr(expr, ctx, self)?;
```

Comment:
Here if I want to convert a `LogicalExprNode` to a `Expr`, I have to call `parse_expr`.
However, `parse_expr` will call `codec. try_decode_udf`, which lead circle call. 
That makes me a little confused how to implement this method.